qiskit/backends/_sympy_qasmsimulator.py

```diff
@@ -0,0 +1,438 @@
+        job_id = str(uuid.uuid4())
+        qobj = q_job.qobj
+        result_list = []
+        self._shots = qobj['config']['shots']
+        if self._shots > 1:
+            print("Warning: no need for multiple shots! set shot=1 automatically!")
+        for circuit in qobj['circuits']:
+            result_list.append(self.run_circuit(circuit))
+        return Result({'job_id': job_id, 'result': result_list, 'status': 'COMPLETED'}, qobj)
+
+    def compute_distribution(self):
+        """compute the distribution and return it in the list form"""
+        matrix_form = represent(self._quantum_state)
+        shapeN = matrix_form.shape[0]
+        list_form = [SympyQasmSimulator._conjugate_square(matrix_form[i, 0]) for i in range(shapeN)]
+        return list_form
+
+    def run_circuit(self, circuit):
+        """Run a circuit and return object
+        Args:
+            circuit (dict): JSON circuit from qobj circuits list
+        Returns:
+            dict: A dictionary of results which looks something like::
+                {
+                "data":{
+                    "time"  : xx.xxxxxxxx
+                    },
+                "status": --status (string)--
+                }
+        Raises:
+            SimulatorError: if an error occurred.
+        """
+        ccircuit = circuit['compiled_circuit']
+        self._number_of_qubits = ccircuit['header']['number_of_qubits']
+        self._number_of_cbits = ccircuit['header']['number_of_clbits']
+        self._quantum_state = 0
+        self._classical_state = 0
+        cl_reg_index = []  # starting bit index of classical register
+        cl_reg_nbits = []  # number of bits in classical register
+        cbit_index = 0
+        for cl_reg in ccircuit['header']['clbit_labels']:
+            cl_reg_nbits.append(cl_reg[1])
+            cl_reg_index.append(cbit_index)
+            cbit_index += cl_reg[1]
+        if circuit['config']['seed'] is None:
+            random.seed(random.getrandbits(32))
+        else:
+            random.seed(circuit['config']['seed'])
+
+        actual_shots = self._shots
+        self._quantum_state = Qubit(*tuple([0]*self._number_of_qubits))
+        self._classical_state = 0
+        # Do each operation in this shot
+        for operation in ccircuit['operations']:
+            if 'conditional' in operation:  # not related to sympy
+                mask = int(operation['conditional']['mask'], 16)
+                if mask > 0:
+                    value = self._classical_state & mask
+                    while (mask & 0x1) == 0:
+                        mask >>= 1
+                        value >>= 1
+                    if value != int(operation['conditional']['val'], 16):
+                        continue
+            if operation['name'] in ['U', 'u1', 'u2', 'u3']:
+                qubit = operation['qubits'][0]
+                opname = operation['name'].upper()
+                opparas = operation['params']
```

Comment:
Maybe makes sense to call `regulate` here. In this way, sympifing the parameters here also might simplify `regulate`.

Reply:
regulate() is for fixing the precision loss of the parameters carried by ugates specifically.
so it would be better to leave the calls of regulate() in compute_ugate_matrix().